url7a69e77641_benmouh_MySgbd_tgz-pJ8-Demo_CheckRowNumJ8/src/org/sgbd/Mysgbd/parser/MysgbdParser.java

```diff
@@ -0,0 +1,3669 @@
+/* Generated By:JavaCC: Do not edit this line. MysgbdParser.java */
+package org.sgbd.Mysgbd.parser;
+
+import java.io.*;
+import java.util.*;
+
+import org.sgbd.Mysgbd.*;
+
+
+/**
+ * MySgbdParser: an SQL parser
+ */
+public class MysgbdParser implements MysgbdParserConstants {
+
+  public static void main(String args[]) throws ParseException {
+
+    MysgbdParser p = null ;
+
+    if ( args.length < 1  ) {
+      System.out.println("Reading from stdin (exit; to finish)");
+      p = new MysgbdParser(System.in) ;
+
+    } else {
+
+      try {
+        p = new MysgbdParser(new DataInputStream(
+                          new FileInputStream(args[0]))) ;
+      } catch (FileNotFoundException e) {
+        System.out.println("File " + args[0] +
+                           " not found. Reading from stdin") ;
+        p = new MysgbdParser(System.in) ;
+      }
+    } // else ends here
+
+    if ( args.length > 0 ) {
+      System.out.println(args[0]) ;
+    }
+
+    MStatement st = null;
+    while((st = p.SQLStatement()) != null) {
+      System.out.println(st.toString());
+    }
+/*
+    Vector v = p.SQLStatements();
+
+    for(int i=0; i<v.size(); i++) {
+      MStatement st = (MStatement)v.elementAt(i);
+      System.out.println(st.toString());
+    }
+*/
+
+    System.out.println("Parse Successful") ;
+
+  }
+
+  static final public void BasicDataTypeDeclaration() throws ParseException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 64 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.